SimTransport/PPSProtonTransport/src/HectorTransport.cc

```diff
@@ -130,38 +127,30 @@ bool HectorTransport::transportProton(const HepMC::GenParticle* gpart)
                       break;
      }
      // insert protection for NULL beamlines here
-     h_p->computePath(&*_beamline);
-     is_stop = h_p->stopped(&*_beamline);
+     h_p.computePath(&*_beamline);
+     is_stop = h_p.stopped(&*_beamline);
      if(m_verbosity) LogDebug("HectorTransportEventProcessing") << "HectorTransport:filterPPS: barcode = "
              << line << " is_stop=  "<< is_stop;
-     if (p_out) {
-       delete p_out;
-       p_out = nullptr;
-     } 
      if (is_stop) {
-       delete h_p;
-       h_p=nullptr;
        return false;
      }
      //
      //propagating
      //
-     h_p->propagate( _targetZ );
+     h_p.propagate( _targetZ );
 
-     p_out = new TLorentzVector(PPSTools::HectorParticle2LorentzVector(*h_p,direction));
+     p_out = PPSTools::HectorParticle2LorentzVector(h_p,direction);
 
-     p_out->SetPx(direction*p_out->Px());
-     x1_ctpps = direction*h_p->getX()*um_to_mm; 
-     y1_ctpps = h_p->getY()*um_to_mm;
+     p_out.SetPx(direction*p_out.Px());
+     x1_ctpps = direction*h_p.getX()*um_to_mm; 
+     y1_ctpps = h_p.getY()*um_to_mm;
 
      if(m_verbosity) LogDebug("HectorTransportEventProcessing") <<
              "HectorTransport:filterPPS: barcode = " << line << " x=  "<< x1_ctpps <<" y= " << y1_ctpps;
          
      m_beamPart[line]    = p_out;
      m_xAtTrPoint[line]  = x1_ctpps;
      m_yAtTrPoint[line]  = y1_ctpps;
-     delete p_out; p_out = nullptr;
```

Comment:
This delete was invalidating the value of `m_beamPart[line].second` since the object has been deleted. This was then later deleted again in `ProtonTransport::clear()`.